includes/class-wc-gateway-swedbank-pay-cc.php

```diff
@@ -1194,12 +1201,20 @@ public function return_handler() {
 			$order_id  = absint(  wc_clean( $_GET['order_id'] ) ); // WPCS: input var ok, CSRF ok.
 			$order_key = empty( $_GET['key'] ) ? '' : wc_clean( wp_unslash( $_GET['key'] ) ); // WPCS: input var ok, CSRF ok.
 
-			if ( empty( $order_id ) || empty( $order_id ) ) {
-				throw new Exception( 'An order ID or order key wasn\'t provided' );
+			$order = wc_get_order( $order_id );
+			if ( ! $order ) {
+				throw new Exception( 'Unable to load an order.' );
+			}
+
+			if ( empty( $order_key ) ) {
+				if ( ! in_array( $_SERVER['REMOTE_ADDR'], $this->gateway_ip_addresses ) ) {
+					throw new Exception( 'An order key wasn\'t provided' );
+				}
+
+				$order_key = $order->get_order_key();
```

Comment:
This algorithm does not do what I describe still, @aait. Is there a reason to process the callback at all if the `REMOTE_ADDR` is different to `gateway_ip_addresses`? Can't we do it like this instead? Even better, move the `REMOTE_ADDR` check all the way to the top and return before doing anything else, so the callback endpoint is less prone to DDOS attacks?

```suggestion
            if ( ! in_array( $_SERVER['REMOTE_ADDR'], $this->gateway_ip_addresses ) ) {
                $this->core->log(
                    LogLevel::INFO,
                    sprintf('Incoming Callback from %s ignored as its origin is not PayEx.', $_SERVER['REMOTE_ADDR'])
                return;
            }

            if ( empty( $order_key ) ) {
                $order_key = $order->get_order_key();
```

Reply:
@asbjornu We will have problems with customers who have problems with server configuration including `REMOTE_ADDR`. Some server configurations returns proxy ip instead of real ip. See https://httpd.apache.org/docs/current/mod/mod_remoteip.html

Reply:
@asbjornu I suggest to reject this PR then

Reply:
I see. That makes sense. Thanks.